compiler/ballerina-lang/src/main/java/io/ballerina/projects/CodeModifierManager.java

```diff
@@ -0,0 +1,437 @@
+            try {
+                modifierTask.modify(sourceModifierContext);
+            } catch (Throwable e) {
+                // Used Throwable here catch any sort of error produced by the third-party compiler plugin code
+                String message;
+                if (compilerPluginInfo.kind().equals(CompilerPluginKind.PACKAGE_PROVIDED)) {
+                    PackageProvidedCompilerPluginInfo pkgProvidedCompilerPluginInfo =
+                            (PackageProvidedCompilerPluginInfo) compilerPluginInfo;
+                    PackageDescriptor pkgDesc = pkgProvidedCompilerPluginInfo.packageDesc();
+                    message = "The compiler extension in package '" +
+                            pkgDesc.org() +
+                            ":" + pkgDesc.name() +
+                            ":" + pkgDesc.version() + "' failed to complete. ";
+                } else {
+                    message = "The compiler extension '" + compilerPluginInfo.compilerPlugin().getClass().getName()
+                            + "' failed to complete. ";
+                }
+                throw new ProjectException(message + e.getMessage(), e);
+            }
+        }
+    }
+
+    /**
+     * The default implementation of the {@code CodeModifierContextImpl}.
+     *
+     * @since 2201.0.3
+     */
+    private static class CodeModifierContextImpl implements CodeModifierContext {
+
+        private final CodeModifierTasks codeModifierTasks;
+        private final CodeModifierInfo codeModifierInfo;
+
+        CodeModifierContextImpl(CodeModifierInfo codeModifierInfo,
+                                CodeModifierTasks codeModifierTasks) {
+            this.codeModifierInfo = codeModifierInfo;
+            this.codeModifierTasks = codeModifierTasks;
+        }
+
+        @Override
+        public void addSourceModifierTask(ModifierTask<SourceModifierContext> modifierTask) {
+            codeModifierTasks.addSourceModifierTask(codeModifierInfo,
+                    new SourceModifierTask(modifierTask, codeModifierInfo.compilerPluginInfo));
+        }
+
+        @Override
+        public void addSyntaxNodeAnalysisTask(AnalysisTask<SyntaxNodeAnalysisContext> analysisTask,
+                                              SyntaxKind syntaxKind) {
+            addSyntaxNodeAnalysisTask(analysisTask, Collections.singletonList(syntaxKind));
+        }
+
+        @Override
+        public void addSyntaxNodeAnalysisTask(AnalysisTask<SyntaxNodeAnalysisContext> analysisTask,
+                                              Collection<SyntaxKind> syntaxKinds) {
+            codeModifierTasks.addSyntaxNodeAnalysisTask(codeModifierInfo,
+                    new SyntaxNodeAnalysisTask(analysisTask, syntaxKinds, codeModifierInfo.compilerPluginInfo));
+        }
+    }
+
+
+    /**
+     * A container that maintain various code modifier tasks against the {@code CodeModifier} instance.
+     *
+     * @since 2.0.0
```

Comment:
```suggestion
     * @since 2201.1.0
```
Please correct other places as well

Reply:
Added